### DISCUSSION TITLE
How can I have a task that does many things at once?

### DISCUSSION THREAD

**Discussion by Zeioth** [stevearc/overseer.nvim#149] [General]
I want that when I run 

    OverseerRun

my task is listed. But I need my task to run several commands an once. How can achieve this?

**Comment by stevearc** [comment]
There are two ways to run multiple tasks. One is using the "dependencies" component, and one (as you have found) is the orchestrator strategy. The docs are available here: https://github.com/stevearc/overseer.nvim/blob/master/doc/guides.md#running-tasks-sequentially

**Reply by Zeioth** [reply]
I understand that the scenario to use dependency is when a task depend on another and you want to be sure they run secuentially.

EDIT: Actually, this can also be done on the orchestrator strategy. Isn't it the dependency strategy redundant?

**Reply by stevearc** [reply]
You can get the same or similar behavior with the orchestrator strategy, but the advantage of the dependency component is that it's a bit lighter weight and (debatably) easier to use. The orchestrator strategy also requires a single, top-level task that handles the running of all other tasks. The dependency component will simply cause the task at hand to run other tasks first. They are small differences, but it's useful to have both.
